lesson21/src/js/main.js

```diff
@@ -0,0 +1,212 @@
+    }
+}
+
+const sortButtons = [
+    {dataOrder:"asc",src:"../img/asc.svg", alt:"asc-image"},
+    {dataOrder:"desc",src:"../img/desc.svg",alt:"desc-image"},
+    {dataOrder:"both",src:"../img/both.svg",alt:"both-image"}
+]
+
+const sortByUsersTableColumn = {
+    "id" : "ID"
+}
+
+const addSortButtons = (userKey,userValue,th) => {
+    for(const value of Object.values(sortByUsersTableColumn)){
+        userValue === value && createSortButtons(userKey,th) ;
+    }
+}
+
+const createSortButtons = (columnKey,th) => {
+    const sortButtonsBox = createAttributedElements ({
+        tag:"div",
+        valuesByAttributes:{
+            id:`js-${columnKey}Buttons-Box`,
+            class: "w-10 inline-block align-middle"
+        }
+    })
+
+    const fragment = document.createDocumentFragment();
+
+    for(const button of sortButtons){
+        const sortButton = createAttributedElements({
+            tag:"button",
+            valuesByAttributes:{
+                class: `js-${columnKey}SortButton`,
+                'data-order':button.dataOrder
+            }
+        });
+
+        button.dataOrder !== "both" && sortButton.classList.add("hidden");
+
+        const sortButtonImage = createAttributedElements({
+            tag:"img",
+            valuesByAttributes:{
+                class:"pointer-events-none",
+                src:button.src,
+                alt:button.alt
+            }
+        })
+
+        fragment.appendChild(sortButton).appendChild(sortButtonImage);
+    }
+    th.appendChild(sortButtonsBox).appendChild(fragment);
+}
+
+const toggleHiddenClassSortButton = (sortButtons,index) => {
+    sortButtons[index].classList.add("hidden");
+    (index === sortButtons.length -1) ? index = 0 : index++;
+    sortButtons[index].classList.remove("hidden");
+}
+
+const upDateTableBody = usersData =>  {
+    document.getElementById("js-usersTableBody").remove();
+    renderTableBody(usersData);
+};
+
+const sortUsersData = (key, usersData, dataOrder) => {
+    const cloneUsersData = [...usersData];
+
+    const sortButtonsFunc = {
+        "both": () => {
+            cloneUsersData.sort((a,b) => a[key] - b[key]);
+            upDateTableBody(cloneUsersData);
+        },
+        "asc" : () => {
+            cloneUsersData.sort((a,b) => b[key] - a[key]);
+            upDateTableBody(cloneUsersData);
+        },
+        "desc": () => {
+            upDateTableBody(usersData);
+        }
+    }
+
+    Object.keys(sortButtonsFunc).forEach((button) => dataOrder === button && sortButtonsFunc[button]() );
+}
+
+const sortTableBody = usersData => {
+    for(const key of Object.keys(sortByUsersTableColumn)){
+        const sortButtons = [...document.getElementsByClassName(`js-${key}SortButton`)];
+        const sortButtonsBox = document.getElementById(`js-${key}Buttons-Box`);
+        
+        let dataOrder;
+        sortButtonsBox.addEventListener("click", (e) => {
+            let index = sortButtons.indexOf(e.target);
```

Comment:
IMO
I will share the review that my teacher gave me.
The name `index` leaves out too much information.
I think it would be easier to understand if you wrote in detail.

Reply:
I didn't give it a specific name because the purpose here is to know the index number of the element stored in the sortButtons array, but what other name would be better?

The name I came up with is below, but it seems redundant.
Any advice would be appreciated!

```JavaScript
const sortButtonIndex = sortButtons.indexOf(e.target);
```

Reply:
Thank you for your comment!
I received a review on the following code for assignment 26.
It was in a similar addEventLister.
How about this?

```
// lesson16
tabNav.addEventListener("click", (e) => {
    const clickedTabIndex = e.target.dataset.index;
```

Reply:
Thank you for your very good advice.
It was very helpful!
I fixed.
https://github.com/nakagawamai/js-lesson/pull/23/commits/068ee28f9c019c7b254ea7d78538f77598ccd403